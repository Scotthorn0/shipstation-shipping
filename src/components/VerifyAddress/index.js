import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Segment, Form, Dimmer, Loader, Header } from 'semantic-ui-react';
import { NEXT } from '../../constants';

class VerifyAddress extends Component {
  static propTypes = {
    requestAddressVerification: PropTypes.func,
    verifyingAddress: PropTypes.bool,
    addressVerified: PropTypes.bool,
    moveStep: PropTypes.func,
  }

  state = {
    name: '',
    phone: '',
    company: '',
    address: '',
    city: '',
    state: '',
    postal: '',
    country: '',
  }

  componentWillReceiveProps({ addressVerified }) {
    const { moveStep } = this.props;
    if (addressVerified) {
      moveStep(NEXT);
    }
  }

  onSubmit = (_, e) => {
    const { requestAddressVerification } = this.props;

    requestAddressVerification(this.state);
  }

  onChange = (_, { name, value }) => {
    this.setState({ [name]: value });
  }

  render() {
    const {
      name, phone, company, address, city, state, postal, country,
    } = this.state;
    const { verifyingAddress } = this.props;

    return (
      <Segment>
        <Dimmer active={verifyingAddress}>
          <Loader />
        </Dimmer>
        <Header content="Where are we sending this beautiful plant?" subheader="Enter your address below" />
        <Form onSubmit={this.onSubmit}>
          <Form.Input
            label="Name"
            required
            type="text"
            autoComplete="name"
            name="name"
            icon="user"
            placeholder="Full Name"
            value={name}
            onChange={this.onChange}
          />
          <Form.Input
            label="Telephone"
            type="tel"
            autoComplete="tel"
            name="phone"
            icon="phone"
            placeholder="Area Code and Phone Number"
            value={phone}
            onChange={this.onChange}
          />
          <Form.Input
            label="Company"
            type="text"
            name="company"
            icon="building"
            placeholder="Company Name"
            value={company}
            onChange={this.onChange}
          />
          <Form.Input
            label="Street Address"
            required
            autoComplete="street-address"
            name="address"
            icon="address card"
            placeholder="Address"
            value={address}
            onChange={this.onChange}
          />
          <Form.Input
            label="City"
            required
            autoComplete="address-level2"
            name="city"
            placeholder="City"
            value={city}
            onChange={this.onChange}
          />
          <Form.Input
            label="State"
            required
            autoComplete="address-level1"
            name="state"
            placeholder="State"
            value={state}
            onChange={this.onChange}
          />
          <Form.Input
            label="Postal Code"
            required
            autoComplete="postal_code"
            name="postal"
            placeholder="Zip or Postal Code"
            value={postal}
            onChange={this.onChange}
          />
          <Form.Input
            label="Country"
            required
            autoComplete="country"
            name="country"
            placeholder="Country"
            value={country}
            onChange={this.onChange}
          />
          <Form.Button>Check Address</Form.Button>
        </Form>
      </Segment>
    );
  }
}

export default VerifyAddress;
