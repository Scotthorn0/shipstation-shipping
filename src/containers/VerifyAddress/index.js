import { connect } from 'react-redux';
import VerifyComponent from '../../components/VerifyAddress';
import { requestAddressVerification } from '../../actions';

export const mapStateToProps = ({ verifyingAddress, addressVerified }) => ({
  verifyingAddress,
  addressVerified,
});

export default connect(mapStateToProps, { requestAddressVerification })(VerifyComponent);
