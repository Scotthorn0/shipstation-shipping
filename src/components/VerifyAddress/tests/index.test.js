import React from 'react';
import { shallow } from 'enzyme';
import VerifyAddress from '../';

describe('VerifyAddress Component', () => {
  let wrapper;
  let instance;
  let requestAddressVerificationMock;
  let moveStepMock;

  beforeEach(() => {
    requestAddressVerificationMock = jest.fn();
    moveStepMock = jest.fn();
    wrapper = shallow(
      <VerifyAddress
        moveStep={moveStepMock}
        requestAddressVerification={requestAddressVerificationMock}
      />,
    );
    instance = wrapper.instance();
  });

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe.true;
  });

  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('componentWillReceiveProps', () => {
    test('should not call moveStep if the address is not verified', () => {
      instance.componentWillReceiveProps({ addressVerified: false });
      expect(moveStepMock.called).toBe.false;
    });
    test('should call moveStep if the address is verified', () => {
      instance.componentWillReceiveProps({ addressVerified: true });
      expect(moveStepMock.called).toBe.true;
    });
  });

  describe('onSubmit', () => {
    test('should call requestAddressVerification with the values in state', () => {
      const state = {
        name: 'Mickey and Minnie Mouse',
        phone: '714-781-4565',
        company: 'The Walt Disney Company',
        address: '500 South Buena Vista Street',
        city: 'Burbank',
        state: 'CA',
        postal: '91521',
        country: 'US',
      };
      wrapper.setState(state);
      instance.onSubmit();
      expect(requestAddressVerificationMock.mock.calls[0][0]).toEqual(state);
    });
  });

  describe('onChange', () => {
    test('should update state with the correct key and value', () => {
      instance.onChange({}, { name: 'address', value: '500 South Buena Vista Street' });
      expect(wrapper.state('address')).toBe('500 South Buena Vista Street');
    });
  });
});
