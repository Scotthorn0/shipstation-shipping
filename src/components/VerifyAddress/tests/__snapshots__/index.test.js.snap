// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyAddress Component should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifyAddress
    moveStep={[Function]}
    requestAddressVerification={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Dimmer
          active={undefined}
>
          <Loader />
</Dimmer>,
        <Header
          content="Where are we sending this beautiful plant?"
          subheader="Enter your address below"
/>,
        <Form
          as="form"
          onSubmit={[Function]}
>
          <FormInput
                    as={[Function]}
                    autoComplete="name"
                    control={[Function]}
                    icon="user"
                    label="Name"
                    name="name"
                    onChange={[Function]}
                    placeholder="Full Name"
                    required={true}
                    type="text"
                    value=""
          />
          <FormInput
                    as={[Function]}
                    autoComplete="tel"
                    control={[Function]}
                    icon="phone"
                    label="Telephone"
                    name="phone"
                    onChange={[Function]}
                    placeholder="Area Code and Phone Number"
                    type="tel"
                    value=""
          />
          <FormInput
                    as={[Function]}
                    control={[Function]}
                    icon="building"
                    label="Company"
                    name="company"
                    onChange={[Function]}
                    placeholder="Company Name"
                    type="text"
                    value=""
          />
          <FormInput
                    as={[Function]}
                    autoComplete="street-address"
                    control={[Function]}
                    icon="address card"
                    label="Street Address"
                    name="address"
                    onChange={[Function]}
                    placeholder="Address"
                    required={true}
                    value=""
          />
          <FormInput
                    as={[Function]}
                    autoComplete="address-level2"
                    control={[Function]}
                    label="City"
                    name="city"
                    onChange={[Function]}
                    placeholder="City"
                    required={true}
                    value=""
          />
          <FormInput
                    as={[Function]}
                    autoComplete="address-level1"
                    control={[Function]}
                    label="State"
                    name="state"
                    onChange={[Function]}
                    placeholder="State"
                    required={true}
                    value=""
          />
          <FormInput
                    as={[Function]}
                    autoComplete="postal_code"
                    control={[Function]}
                    label="Postal Code"
                    name="postal"
                    onChange={[Function]}
                    placeholder="Zip or Postal Code"
                    required={true}
                    value=""
          />
          <FormInput
                    as={[Function]}
                    autoComplete="country"
                    control={[Function]}
                    label="Country"
                    name="country"
                    onChange={[Function]}
                    placeholder="Country"
                    required={true}
                    value=""
          />
          <FormButton
                    as={[Function]}
                    control={[Function]}
          >
                    Check Address
          </FormButton>
</Form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": undefined,
          "children": <Loader />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "content": "Where are we sending this beautiful plant?",
          "subheader": "Enter your address below",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "form",
          "children": Array [
            <FormInput
              as={[Function]}
              autoComplete="name"
              control={[Function]}
              icon="user"
              label="Name"
              name="name"
              onChange={[Function]}
              placeholder="Full Name"
              required={true}
              type="text"
              value=""
/>,
            <FormInput
              as={[Function]}
              autoComplete="tel"
              control={[Function]}
              icon="phone"
              label="Telephone"
              name="phone"
              onChange={[Function]}
              placeholder="Area Code and Phone Number"
              type="tel"
              value=""
/>,
            <FormInput
              as={[Function]}
              control={[Function]}
              icon="building"
              label="Company"
              name="company"
              onChange={[Function]}
              placeholder="Company Name"
              type="text"
              value=""
/>,
            <FormInput
              as={[Function]}
              autoComplete="street-address"
              control={[Function]}
              icon="address card"
              label="Street Address"
              name="address"
              onChange={[Function]}
              placeholder="Address"
              required={true}
              value=""
/>,
            <FormInput
              as={[Function]}
              autoComplete="address-level2"
              control={[Function]}
              label="City"
              name="city"
              onChange={[Function]}
              placeholder="City"
              required={true}
              value=""
/>,
            <FormInput
              as={[Function]}
              autoComplete="address-level1"
              control={[Function]}
              label="State"
              name="state"
              onChange={[Function]}
              placeholder="State"
              required={true}
              value=""
/>,
            <FormInput
              as={[Function]}
              autoComplete="postal_code"
              control={[Function]}
              label="Postal Code"
              name="postal"
              onChange={[Function]}
              placeholder="Zip or Postal Code"
              required={true}
              value=""
/>,
            <FormInput
              as={[Function]}
              autoComplete="country"
              control={[Function]}
              label="Country"
              name="country"
              onChange={[Function]}
              placeholder="Country"
              required={true}
              value=""
/>,
            <FormButton
              as={[Function]}
              control={[Function]}
>
              Check Address
</FormButton>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "name",
              "control": [Function],
              "icon": "user",
              "label": "Name",
              "name": "name",
              "onChange": [Function],
              "placeholder": "Full Name",
              "required": true,
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "tel",
              "control": [Function],
              "icon": "phone",
              "label": "Telephone",
              "name": "phone",
              "onChange": [Function],
              "placeholder": "Area Code and Phone Number",
              "type": "tel",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "control": [Function],
              "icon": "building",
              "label": "Company",
              "name": "company",
              "onChange": [Function],
              "placeholder": "Company Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "street-address",
              "control": [Function],
              "icon": "address card",
              "label": "Street Address",
              "name": "address",
              "onChange": [Function],
              "placeholder": "Address",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "address-level2",
              "control": [Function],
              "label": "City",
              "name": "city",
              "onChange": [Function],
              "placeholder": "City",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "address-level1",
              "control": [Function],
              "label": "State",
              "name": "state",
              "onChange": [Function],
              "placeholder": "State",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "postal_code",
              "control": [Function],
              "label": "Postal Code",
              "name": "postal",
              "onChange": [Function],
              "placeholder": "Zip or Postal Code",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "autoComplete": "country",
              "control": [Function],
              "label": "Country",
              "name": "country",
              "onChange": [Function],
              "placeholder": "Country",
              "required": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": [Function],
              "children": "Check Address",
              "control": [Function],
            },
            "ref": null,
            "rendered": "Check Address",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Dimmer
            active={undefined}
>
            <Loader />
</Dimmer>,
          <Header
            content="Where are we sending this beautiful plant?"
            subheader="Enter your address below"
/>,
          <Form
            as="form"
            onSubmit={[Function]}
>
            <FormInput
                        as={[Function]}
                        autoComplete="name"
                        control={[Function]}
                        icon="user"
                        label="Name"
                        name="name"
                        onChange={[Function]}
                        placeholder="Full Name"
                        required={true}
                        type="text"
                        value=""
            />
            <FormInput
                        as={[Function]}
                        autoComplete="tel"
                        control={[Function]}
                        icon="phone"
                        label="Telephone"
                        name="phone"
                        onChange={[Function]}
                        placeholder="Area Code and Phone Number"
                        type="tel"
                        value=""
            />
            <FormInput
                        as={[Function]}
                        control={[Function]}
                        icon="building"
                        label="Company"
                        name="company"
                        onChange={[Function]}
                        placeholder="Company Name"
                        type="text"
                        value=""
            />
            <FormInput
                        as={[Function]}
                        autoComplete="street-address"
                        control={[Function]}
                        icon="address card"
                        label="Street Address"
                        name="address"
                        onChange={[Function]}
                        placeholder="Address"
                        required={true}
                        value=""
            />
            <FormInput
                        as={[Function]}
                        autoComplete="address-level2"
                        control={[Function]}
                        label="City"
                        name="city"
                        onChange={[Function]}
                        placeholder="City"
                        required={true}
                        value=""
            />
            <FormInput
                        as={[Function]}
                        autoComplete="address-level1"
                        control={[Function]}
                        label="State"
                        name="state"
                        onChange={[Function]}
                        placeholder="State"
                        required={true}
                        value=""
            />
            <FormInput
                        as={[Function]}
                        autoComplete="postal_code"
                        control={[Function]}
                        label="Postal Code"
                        name="postal"
                        onChange={[Function]}
                        placeholder="Zip or Postal Code"
                        required={true}
                        value=""
            />
            <FormInput
                        as={[Function]}
                        autoComplete="country"
                        control={[Function]}
                        label="Country"
                        name="country"
                        onChange={[Function]}
                        placeholder="Country"
                        required={true}
                        value=""
            />
            <FormButton
                        as={[Function]}
                        control={[Function]}
            >
                        Check Address
            </FormButton>
</Form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": undefined,
            "children": <Loader />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "content": "Where are we sending this beautiful plant?",
            "subheader": "Enter your address below",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "form",
            "children": Array [
              <FormInput
                as={[Function]}
                autoComplete="name"
                control={[Function]}
                icon="user"
                label="Name"
                name="name"
                onChange={[Function]}
                placeholder="Full Name"
                required={true}
                type="text"
                value=""
/>,
              <FormInput
                as={[Function]}
                autoComplete="tel"
                control={[Function]}
                icon="phone"
                label="Telephone"
                name="phone"
                onChange={[Function]}
                placeholder="Area Code and Phone Number"
                type="tel"
                value=""
/>,
              <FormInput
                as={[Function]}
                control={[Function]}
                icon="building"
                label="Company"
                name="company"
                onChange={[Function]}
                placeholder="Company Name"
                type="text"
                value=""
/>,
              <FormInput
                as={[Function]}
                autoComplete="street-address"
                control={[Function]}
                icon="address card"
                label="Street Address"
                name="address"
                onChange={[Function]}
                placeholder="Address"
                required={true}
                value=""
/>,
              <FormInput
                as={[Function]}
                autoComplete="address-level2"
                control={[Function]}
                label="City"
                name="city"
                onChange={[Function]}
                placeholder="City"
                required={true}
                value=""
/>,
              <FormInput
                as={[Function]}
                autoComplete="address-level1"
                control={[Function]}
                label="State"
                name="state"
                onChange={[Function]}
                placeholder="State"
                required={true}
                value=""
/>,
              <FormInput
                as={[Function]}
                autoComplete="postal_code"
                control={[Function]}
                label="Postal Code"
                name="postal"
                onChange={[Function]}
                placeholder="Zip or Postal Code"
                required={true}
                value=""
/>,
              <FormInput
                as={[Function]}
                autoComplete="country"
                control={[Function]}
                label="Country"
                name="country"
                onChange={[Function]}
                placeholder="Country"
                required={true}
                value=""
/>,
              <FormButton
                as={[Function]}
                control={[Function]}
>
                Check Address
</FormButton>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "name",
                "control": [Function],
                "icon": "user",
                "label": "Name",
                "name": "name",
                "onChange": [Function],
                "placeholder": "Full Name",
                "required": true,
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "tel",
                "control": [Function],
                "icon": "phone",
                "label": "Telephone",
                "name": "phone",
                "onChange": [Function],
                "placeholder": "Area Code and Phone Number",
                "type": "tel",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "control": [Function],
                "icon": "building",
                "label": "Company",
                "name": "company",
                "onChange": [Function],
                "placeholder": "Company Name",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "street-address",
                "control": [Function],
                "icon": "address card",
                "label": "Street Address",
                "name": "address",
                "onChange": [Function],
                "placeholder": "Address",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "address-level2",
                "control": [Function],
                "label": "City",
                "name": "city",
                "onChange": [Function],
                "placeholder": "City",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "address-level1",
                "control": [Function],
                "label": "State",
                "name": "state",
                "onChange": [Function],
                "placeholder": "State",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "postal_code",
                "control": [Function],
                "label": "Postal Code",
                "name": "postal",
                "onChange": [Function],
                "placeholder": "Zip or Postal Code",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "autoComplete": "country",
                "control": [Function],
                "label": "Country",
                "name": "country",
                "onChange": [Function],
                "placeholder": "Country",
                "required": true,
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": [Function],
                "children": "Check Address",
                "control": [Function],
              },
              "ref": null,
              "rendered": "Check Address",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
